centralstore/redis_store.go

```diff
@@ -325,19 +325,52 @@ func (r *RedisBasicStore) GetStatusForTraces(ctx context.Context, traceIDs []str
 	}
 
 	validStates := make(map[CentralTraceState]struct{}, len(statesToCheck))
+	var decisionMade bool
 	for _, state := range statesToCheck {
 		validStates[state] = struct{}{}
+		if (state == DecisionKeep || state == DecisionDrop) && !decisionMade {
+			decisionMade = true
+		}
```

Comment:
```suggestion
		// is any of the states we're looking for a Decision state?
		decisionMade |= (state == DecisionKeep || state == DecisionDrop)
```
I find this clearer. Do you?